datasets/bnl_newspapers/bnl_newspapers.py

```diff
@@ -0,0 +1,188 @@
+                "text": datasets.Value("string"),
+                "pub_date": datasets.Value("timestamp[s]"),
+                "publisher": datasets.Value("string"),
+                "language": datasets.Value("string"),
+                "article_type": datasets.ClassLabel(
+                    names=[
+                        "ADVERTISEMENT_SECTION",
+                        "BIBLIOGRAPHY",
+                        "CHAPTER",
+                        "INDEX",
+                        "CONTRIBUTION",
+                        "TABLE_OF_CONTENTS",
+                        "WEATHER",
+                        "SHIPPING",
+                        "SECTION",
+                        "ARTICLE",
+                        "TITLE_SECTION",
+                        "DEATH_NOTICE",
+                        "SUPPLEMENT",
+                        "TABLE",
+                        "ADVERTISEMENT",
+                        "CHART_DIAGRAM",
+                        "ILLUSTRATION",
+                        "ISSUE",
+                    ]
+                ),
+                "extent": datasets.Value("int32"),
+            }
+        )
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+            # Citation for the dataset
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        _URL = self.config.data_url
+        data_dir = dl_manager.download_and_extract(_URL)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "dirpath": data_dir,
+                },
+            ),
+        ]
+
+    def _generate_examples(
+        self,
+        dirpath,
+    ):
+        """Yields examples as (key, example) tuples."""
+        ns = {
+            "": "http://www.openarchives.org/OAI/2.0/",
+            "xsi": "http://www.w3.org/2001/XMLSchema-instance",
+            "oai_dc": "http://www.openarchives.org/OAI/2.0/oai_dc/",
+            "dc": "http://purl.org/dc/elements/1.1/",
+            "dcterms": "http://purl.org/dc/terms/",
+        }
+        for id_, xml in enumerate(Path(dirpath).rglob("**/*.xml")):
+            tree = ET.parse(xml)
```

Comment:
we need to open the file for it to work in streaming mode (ET.parse doesn't support URLs, while we extended `open` to support URLs)
```suggestion
            tree = ET.parse(open(xml, encoding="utf-8"))
```

Reply:
Hey @lhoestq, please take into account that we now support `ET.parse` directly in streaming mode, since this PR:
- #3476

This is useful for all community datasets, where we cannot impose our convention of using `xopen`...
:wink: